crates/stackable-operator/src/builder/pod/container.rs

```diff
@@ -188,29 +208,53 @@ impl ContainerBuilder {
         self
     }
 
+    /// This function only adds the [`VolumeMount`] in case there is no volumeMount with the same mountPath already.
+    /// In case there already was a volumeMount with the same path already, an [`tracing::error`] is raised in case the
+    /// contents of the volumeMounts differ.
+    ///
+    /// Historically this function unconditionally added volumeMounts, which resulted in invalid
+    /// [`k8s_openapi::api::core::v1::PodSpec`]s, as volumeMounts where added multiple times - think of Trino using the same [`crate::commons::s3::S3Connection`]
+    /// two times, resulting in e.g. the s3 credentials being mounted twice as the same volumeMount.
+    pub fn add_volume_mount_struct(&mut self, volume_mount: VolumeMount) -> Result<&mut Self> {
```

Comment:
question: Does this function need to be public? We already have two functions which should cover all uses cases. I would like to try to keep the API interface as little as possible.

Another option is to deprecate the other two functions and always require a `VolumeMount` to be passed. That the removes a bunch of easy-of-use.

suggestion: If made private, let's rename the function to `add_volume_mount_impl`.

Reply:
I mostly added this to be consistent with PodBuilder and volume addition. Also docs brake, will take care of that later once we have settled on docs.
Anyway, made it private in a814f41cf79fb7ed825a1fade13b2790221ad023